### PR TITLE
joe review notes

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
@fivetran-sheringuyen absolutely fantastic job putting this source package together! 💯 🙌 🕺 

I ran the package locally and it worked like a charm. I also really appreciated how diligent you were with generating the source documentation. This was a great first source package. Thank you for all your hard work here! I do have a few comments that you can find inline as well as a few action items I have in checkboxes below. Let me know if you have any questions:

- [x] Work through the CircleCi errors that are happening related to the auto generated docs section.
- [x] Can you add a CHANGELOG.md and have a section for the v0.1.0 initial release of the package.
- [x] Can you remove the `analysis` folder. This is not needed for the package.
- [x] Can you remove the `snapshots` folder. This is not needed for the package.
- [x] Can you remove the `tests` folder. This is not needed for the package.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
 [![Apache License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 # Apple Search Ads (Source) 
 
-This package models Jira data from [Fivetran's connector](https://fivetran.com/docs/applications/apple-search-ads). It uses data in the format described by [this ERD](https://fivetran.com/docs/applications/apple-search-ads/#schemainformation).
+This package models Apple Search Ads data from [Fivetran's connector](https://fivetran.com/docs/applications/apple-search-ads). It uses data in the format described by [this ERD](https://fivetran.com/docs/applications/apple-search-ads/#schemainformation).
```
Comment (fivetran-joemarkiewicz) [inline README.md:4]: There are a few stowaways from Jira in the README. Can you make sure to update for Apple Search Ads.

File: README.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-This package contains staging models, designed to work simultaneously with our [Jira modeling package](https://github.com/fivetran/dbt_apple_search_ads).  The staging models:
+This package contains staging models, designed to work simultaneously with our [Apple Search Ads modeling package](https://github.com/fivetran/dbt_apple_search_ads).  The staging models:
 - Remove any rows that are soft-deleted
 - Name columns consistently across all packages:
     - Boolean fields are prefixed with `is_` or `has_`
     - Timestamps are appended with `_at`
-    - ID primary keys are prefixed with the name of the table.  For example, the `issue` table's ID column is renamed `issue_id`.
-    - Foreign keys include the table that they refer to. For example, an issue's `assignee` user ID column is renamed `assignee_user_id`.
+    - ID primary keys are prefixed with the name of the table.  For example, the `issue` table's ID column is renamed `issue_id`. 
+    - Foreign keys include the table that they refer to. For example, an issue's `assignee` user ID column is renamed `assignee_user_id`.
```
Comment (fivetran-joemarkiewicz) [inline README.md:18]: Can you update this to be more specific to Apple Search Ads. (ie. `issue_id` does not exist in Apple Search Ads).

File: models/stg_apple_search_ads__ad_group_history.sql [hunk 3]
```diff
@@ -23,7 +23,7 @@ final as (
     
     select 
         _fivetran_synced,
-        modification_time as modified_at,
+        modification_time as modified_at, --We should check to see if we will need to add logic to get the most recent ad_group
```
Comment (fivetran-joemarkiewicz) [inline models/stg_apple_search_ads__ad_group_history.sql:26]: Typically in these `history` tables, we need to account for records that change. See how we did this in our Snapchat Ads package: 
https://github.com/fivetran/dbt_snapchat_ads_source/blob/main/models/stg_snapchat__ad_squad_history.sql#L34-L37

I have a suspicion we will need to apply similar logic here. We can test this and see if the result replicates into a new record. Or if the modified time simply updates.

File: models/stg_apple_search_ads__campaign_history.sql [hunk 4]
```diff
@@ -23,7 +23,7 @@ final as (
     
     select
         _fivetran_synced,
-        modification_time as modified_at,
+        modification_time as modified_at, --We should check to see if we will need to add logic to get the most recent campaign
```
Comment (fivetran-joemarkiewicz) [inline models/stg_apple_search_ads__campaign_history.sql:26]: Same comment in `models/stg_apple_search_ads__ad_group_history.sql`

File: models/stg_apple_search_ads__keyword_history.sql [hunk 5]
```diff
@@ -23,7 +23,7 @@ final as (
     
     select 
         _fivetran_synced,
-        modification_time as modified_at,
+        modification_time as modified_at, --We should check to see if we will need to add logic to get the most recent keyword
```
Comment (fivetran-joemarkiewicz) [inline models/stg_apple_search_ads__keyword_history.sql:26]: Same comment in `models/stg_apple_search_ads__ad_group_history.sql`

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
See inline comments below

File: models/stg_apple_search_ads__ad_group_history.sql (hunk 3 above)
Comment (fivetran-sheringuyen) [reply models/stg_apple_search_ads__ad_group_history.sql:26]: Ah, I understand that logic! In the case of snapchat, is that to remove duplicates? Or to get the latest set of data for a specific record?

I think the `most_recent` record would make sense if the records are duplicated. But it looks like at least for the `campaign_history` table that new records are created when there are updates to a campaign. Although we only have unique records for the `keyword_history` and `ad_group_history`, I have a hunch that these tables will also follow the same logic.

A use case that I'm thinking:
Specifically for the `campaign_history` table, it looks like this table tracks changes to the record. E.g. when a campaign status goes from `ENABLED` -> `PAUSE`. It's more like a campaign "events" table if that makes sense. I think in this case, it'd be valuable to have all the records and not just the most recent - especially if, for example, users want to calculate the difference in time between a campaign "TOTAL_BUDGET_EXHAUSTED" and when it was supposed to end. You can see this example in `dbt_package_testing.apple_search_ads.campaign_history` (there's only 3 records).

Comment (fivetran-sheringuyen) [reply models/stg_apple_search_ads__ad_group_history.sql:26]: I'm almost wondering if there should be a 1. campaign "look up" table and 2. campaign "events" table. This logic would extend to ad_group_history & keyword_history as well. 

A little more detail (which can be extended to the other two tables):
1. Campaign look up table should be the "cleaned" table with soft deleted records removed, and previous events removed.
2. Campaign "events" table should have all records, including soft deleted records & previous changes to that campaign.

**Pull request closed without merging**